src/main/java/org/example/StudentRepository.java

```diff
@@ -0,0 +1,145 @@
+                throw new IllegalArgumentException("Last name should not be left empty. Please try again.");
+            }
+        }
+
+        System.out.println("Please enter Student date of birth with the following format yyyy-mm-dd: ");
+        dateOfBirthInput = scanner.nextLine();
+        while (!dateOfBirthValidation) {
+            try {
+                createdStudent.dateOfBirth = LocalDate.parse(dateOfBirthInput);
+                if (createdStudent.dateOfBirth.isAfter(LocalDate.parse("1900-01-01")) ||
+                        createdStudent.dateOfBirth.isBefore(LocalDate.parse("2023-04-25"))) {
+                    dateOfBirthValidation = true;
+                } else {
+                    System.out.println("Student date of birth is not valid. Please retry.");
+                }
+            } catch (Exception e) {
+                throw new DateTimeException("Students DOB must be between 1900-01-01 & 2023-04-25");
+            }
+        }
+
+        System.out.println("Please enter Student gender, M for male / F for female: ");
+        while (!genderValidation) {
+            try {
+                genderInput = scanner.nextLine();
+                switch (genderInput) {
+                    case "M", "F" -> {
+                        genderValidation = true;
+                        createdStudent.gender = genderInput;
+                    }
+                    default -> System.out.println("Please enter a valid gender type.");
+                }
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Gender should not be left empty. Please try again.");
+            }
+        }
+
+        System.out.println("Please enter Student ID: ");
+        while (!IdValidation) {
+            try {
+                createdStudent.ID = Integer.parseInt(scanner.nextLine());
+                IdValidation = true;
+            } catch (NumberFormatException | NullPointerException nfe) {
+                throw new IllegalArgumentException("Error, please enter a number based ID.");
+            }
+        }
+        addStudent(createdStudent);
+    }
+
+    void addStudent(Student student) {
+        try {
+            studentHashMap.put(student.ID, student);
+            System.out.println("Student is being added to the repository.");
+        } catch (RuntimeException iae) {
+            System.out.println("The ID you have entered is not valid. Please enter only digits for the ID field.");
+        }
+    }
+
+    void deleteStudent(Integer studentID) {
+        if (studentHashMap.containsKey(studentID)) {
+            studentHashMap.remove(studentID);
+            System.out.println("The student with the following ID: " + studentID + " is removed.");
+        } else if (studentID == null) {
+            System.out.println("The ID entered must not be null. Please try again.");
+        } else {
+            System.out.println("The ID entered is not linked to any student in the repository.");
```

Comment:
this should be together with the other fail condtion. if (id == null || idWasNotFound)